lib/pymedphys/_dicom/create.py

```diff
@@ -40,13 +40,17 @@ def set_default_transfer_syntax(dataset):
         dataset.is_implicit_VR = True
 
 
-def dicom_dataset_from_dict(input_dict: dict, template_ds=None):
+def dicom_dataset_from_dict(input_dict: dict, ensure_file_meta=False, template_ds=None):
     """Create a pydicom DICOM object from a dictionary"""
     if template_ds is None:
         dataset = pydicom.Dataset()
     else:
         dataset = deepcopy(template_ds)
 
+    if ensure_file_meta:
+        dataset.ensure_file_meta()
+        dataset.fix_meta_info(enforce_standard=False)
```

Comment:
I'm not sure what `enforce_standard` does here. What's the reason for `False`?

Reply:
So this might take some thought/discussion:

If I try to access `ds.pixel_array`, `pydicom` tries to convert PixelData and complains:

```console
E       AttributeError: 'Dataset' object has no attribute 'file_meta'
```

So I thought, okay, is there a ready-made function in `pydicom` to make a suitable `file_meta`? I found `ensure_file_meta()`, which creates an *empty* File Meta in the dataset. I also found `fix_meta_info()` that creates or checks for correct values for transfer syntax and media storage UIDs. I figured these might be useful for a barebones dataset creation tool like `dicom_dataset_from_dict()`.

Having just `ensure_file_meta()` isn't enough, since converting PixelData also requires a `TransferSyntaxUID`:

```console
E       AttributeError: 'FileMetaDataset' object has no attribute 'TransferSyntaxUID'
```

`fix_meta_info()` can create TransferSyntaxUIDs easily enough in this test case, but cannot create MediaStorageUIDs without already set SOPClassUIDs and SOPInstanceUIDs. Setting `enforce_standard=True` treats this limitation strictly and throws an error, whereas `False` does not:

```console
E               ValueError: Missing required File Meta Information elements from 'file_meta':
E                       (0002, 0002) MediaStorageSOPClassUID
E                       (0002, 0003) MediaStorageSOPInstanceUID
```

---

Perhaps it's best that I manually set only the required fields myself in the test function. Was just trying to think a little more generally with my proposed edit to `dicom_dataset_from_dict`

Reply:
Also, I've since discovered that `ensure_file_meta()` is superfluous if we're running `fix_meta_info()`

Reply:
I guess, bottom line: a barebones RTDOSE dataset needs a FileMeta containing a TransferSyntaxUID but does not necessarily need MediaStorageUIDs.

Reply:
> Perhaps it's best that I manually set only the required fields myself in the test function. Was just trying to think a little more generally with my proposed edit to dicom_dataset_from_dict

Yeah, I'd prefer to have `dicom_dataset_from_dict` keep the scope of only doing one thing and trying to do it well. At the end of the day the created dataset has these functions on it that the programmer can subsequently call. No need to have `dicom_dataset_from_dict` run of and do things that are not in its purview. 🙂.

Reply:
Maybe I could add a basebones rtdose dataset creator then? Not just for `fix_meta_info()`, but also to ensure certain required fields are set (e.g. `PhotometricInterpretation="MONOCHROME2"`).

Reply:
> Maybe I could add a basebones rtdose dataset creator then? Not just for `fix_meta_info()`, but also to ensure certain required fields are set (e.g. `PhotometricInterpretation="MONOCHROME2"`).

Yes! 🙂 that'd be perfect and awesome. ... another PR though, feel free to stack it on top of this one if you'd like.